plugins/azure/virtualnetworks/networkGatewaysUsage.js

```diff
@@ -0,0 +1,92 @@
+var async = require('async');
+var helpers = require('../../../helpers/azure/');
+
+module.exports = {
+    title: 'Network Gateways Usage',
+    category: 'Virtual Networks',
+    description: 'Ensures that Virtual Networks are using subnets and network security groups instead of virtual network gateways.',
+    more_info: 'To meet your organization\'s security compliance requirements.',
+    link: 'https://docs.microsoft.com/en-us/azure/vpn-gateway/vpn-gateway-about-vpngateways',
+    recommended_action: 'Configure subnets and network security groups instead of virtual network gateways',
+    apis: ['resourceGroups:list','virtualNetworks:listAll','virtualNetworkGateways:listByResourceGroup'],
+
+    run: function(cache, settings, callback) {
+        var results = [];
+        var source = {};
+        var locations = helpers.locations(settings.govcloud);
+
+        async.each(locations.virtualNetworks, function(location, rcb){
+            var virtualNetworks = helpers.addSource(cache, source, 
+                ['virtualNetworks', 'listAll', location]);
+
+            if (!virtualNetworks) return rcb();
+
+            if (virtualNetworks.err || !virtualNetworks.data) {
+                helpers.addResult(results, 3, 'Unable to query for Virtual Networks: ' + helpers.addError(virtualNetworks), location);
+                return rcb();
+            }
+
+            if (!virtualNetworks.data.length) {
+                helpers.addResult(results, 0, 'No existing Virtual Networks found', location);
+                return rcb();
+            }
+
+            var resourceGroups = helpers.addSource(cache, source, 
+                ['resourceGroups', 'list', location]);
+
+            if (!resourceGroups || resourceGroups.err || !resourceGroups.data) {
+                helpers.addResult(results, 3, 'Unable to query for resource groups: ' + helpers.addError(virtualNetworks), location);
+                return rcb();
+            }
+
+            if (!resourceGroups.data.length) {
+                helpers.addResult(results, 0, 'No existing resource groups found', location);
+                return rcb();
+            }
+
+            let gatewaysList = [];
+            resourceGroups.data.forEach(resourceGroup => {
+                var virtualNetworkGateways = helpers.addSource(cache, source, 
+                    ['virtualNetworkGateways', 'listByResourceGroup', location, resourceGroup.id]);
+                
+                if (virtualNetworkGateways.data.length) {
+                    for (let virtualNetworkGateway of virtualNetworkGateways.data) {
+                        gatewaysList.push(virtualNetworkGateway.id);
+                    }
+                }
+            });
+
```

Comment:
I think we should only check if gateway in use for the context of this plugin. Lets discuss this

Reply:
Yeah we can discuss about this